resources/shaders/rt_hybrid.hlsl

```diff
@@ -218,13 +229,27 @@ void ReflectionHit(inout ReflectionHitInfo payload, in MyAttributes attr)
 	const Vertex v1 = g_vertices[indices.y];
 	const Vertex v2 = g_vertices[indices.z];
 
+	uint use_albedo_constant = material.data.flags & MATERIAL_USE_ALBEDO_CONSTANT;
+	uint has_albedo_texture = material.data.flags & MATERIAL_HAS_ALBEDO_TEXTURE;
+	uint use_roughness_constant = material.data.flags & MATERIAL_USE_ROUGHNESS_CONSTANT;
+	uint has_roughness_texture = material.data.flags & MATERIAL_HAS_ROUGHNESS_TEXTURE;
+	uint use_metallic_constant = material.data.flags & MATERIAL_USE_METALLIC_CONSTANT;
+	uint has_metallic_texture = material.data.flags & MATERIAL_HAS_METALLIC_TEXTURE;
+	uint use_normal_texture = material.data.flags & MATERIAL_HAS_NORMAL_TEXTURE;
+
 	//Get data from VBO
 	float2 uv = HitAttribute(float3(v0.uv, 0), float3(v1.uv, 0), float3(v2.uv, 0), attr).xy;
 	uv.y = 1.0f - uv.y;
 
-	float3 albedo = g_textures[material.albedo_id].SampleLevel(s0, uv, 0).xyz;
-	float roughness = max(0.05, g_textures[material.roughness_id].SampleLevel(s0, uv, 0).x);
-	float metal = g_textures[material.metalicness_id].SampleLevel(s0, uv, 0).x;
+	float3 albedo = lerp(g_textures[material.albedo_id].SampleLevel(s0, uv, 0).xyz,
+		material.data.albedo_alpha.xyz,
+		use_albedo_constant != 0 || has_albedo_texture == 0);
+	float roughness = lerp(max(0.05, g_textures[material.roughness_id].SampleLevel(s0, uv, 0).x),
+		material.data.metallic_roughness.w,
+		use_roughness_constant != 0 || has_roughness_texture == 0);
+	float metal = lerp(g_textures[material.metalicness_id].SampleLevel(s0, uv, 0).x,
+		length(material.data.metallic_roughness.xyz),
+		use_metallic_constant != 0 || has_metallic_texture == 0);
 
```

Comment:
Looks good, consider making the above code, as well as the flags checking code into a separate function that can be reused in all the rendering pipelines. The textures can be passed in as parameters.